src/wallet.js

```diff
@@ -394,6 +394,39 @@ const wallet = {
     }
   },
 
+  /**
+   * Validates old PIN and change it for the new one
+   *
+   * @param {string} oldPin
+   * @param {string} newPin
+   *
+   * @return {boolean} if PIN was changed
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  changePin(oldPin, newPin) {
+    const isCorrect = this.isPinCorrect(oldPin);
+    if (!isCorrect) {
+      return false;
+    }
+
+    const accessData = storage.getItem('wallet:accessData');
+
+    // Get new PIN hash
+    const newHash = this.hashPassword(newPin);
+    // Update new PIN data in storage
+    accessData['hash'] = newHash.key.toString();
+    accessData['salt'] = newHash.salt;
+    storage.setItem('wallet:accessData', accessData);
```

Comment:
You should move this `setItem` to the end of the method, after decrypting/encrypting the `mainKey`.